WIP Setup tool

Daft-Freak opened the pull request:
Helps setting up a new project/ getting the SDK. Probably needs more error handling and tests,

Gadgetoid commented:
This is an excellent idea and appears to work well, nice!

It's a shame the SDK path doesn't tab complete, but that looks to be... rocky territory: https://github.com/pallets/click/issues/173

Some feedback at the very end of the process- ie: "Your new project has been created in: project-dir" would be nice. It sort of trails off after "downloading boilerplate."

```
Project name: pong
Author name: Gadgetoid
32Blit SDK path [/home/phil/32blit-sdk]: /home/phil/Development/32blit/32blit-sdk
Initialise a Git repository? [Y/n]: y
Create VS Code configuration? [Y/n]: y
Create Visual Studio configuration? [y/N]: y
Downloading boilerplate...
```

I also can't begin to imagine how we'd test this using conventional means, but I guess an action could build/install the tools and actually just run the commands directly.

Daft-Freak commented:
Fixed up some more windows stuff (backslashes...) also added the slightly silly VS cmd.exe workaround. Still need to add some error handling, explodes a bit if the SDK/boilerplate git stuff goes wrong for some reason.

Good point on the feedback (I think I intended to do that, but forgot... oops). And yeah, unsure on tests.

Hmm, may need to tweak the "SDK not found" bit, you can currently automate everything else.

Daft-Freak commented:
Ugh, okay that PATH fix only works if you configure _then_ add it since it overwrites the one VS is providing... which provides the tools and no system paths...

It's also supposed to be fixed: https://developercommunity.visualstudio.com/t/cmdexe-is-not-recognized-when-using-cmake-with-nin/367003#T-N429389 (and the provided workaround doesn't work either!)

Gadgetoid commented:
This workflow will test the setup tool and that the output project builds (on Linux, STM32 build) - https://github.com/Gadgetoid/32blit-tools/blob/setup-tool/.github/workflows/test-setup.yml

Doesn't help coveralls, but we can just ignore that.

Gadgetoid commented:
Thank you. I need to coordinate a release now :grimacing:
